docs/src/dev/provider/gremlin-semantics.asciidoc

```diff
@@ -776,6 +776,32 @@ None
 See: link:https://github.com/apache/tinkerpop/tree/x.y.z/gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/dsl/graph/GraphTraversal.java[source],
 link:https://tinkerpop.apache.org/docs/x.y.z/reference/#as-step[reference]
 
+[[asBool-step]]
+=== asBool()
+
+*Description:* Parse the value of the incoming traverser as boolean.
+
+*Syntax:* `asBool()`
+
+[width="100%",options="header"]
+|=========================================================
+|Start Step |Mid Step |Modulated |Domain |Range
+|N |Y |N |`Number`/`String`/`null` |`boolean`
+|=========================================================
+
+*Arguments:*
+
+None
+
+`null` values return `null`, numbers evaluate to `true` if non-zero. Strings only accept "true" or "false", any other strings will return `null`.
+
+*Exceptions*
+
+If the incoming traverser type is unsupported or a string other than "true" or "false" then an `IllegalArgumentException` is thrown.
```

Comment:
For future thoughts - should the semantics doc be Java centric? not sure what it should be for "Exceptions" but might want to do that differently going forward, particularly for 4.x